R/prophet.R

```diff
@@ -0,0 +1,1189 @@
+#
+# Copyright (C) 2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Main function ----
+Prophet <- function(jaspResults, dataset = NULL, options) {
+
+  options <- .prophetInitOptions(jaspResults, options)
+  
+  ready <- (options$dependent != "" && options$time != "")
```

Comment:
just a nitpick: in JASP we prefer subsetting lists like `options[["dependent"]]` rather than with a dollar sign. Not crucial but just so you know.

Reply:
Is there a particular reason for that or is that a style thing? 😃

Reply:
Apart from style, I think the most important reason is to avoid partial matching. I am not sure if there are more reasons to it. But as I said, I don't think it's crucial (I am also guilty of not following it 100%). Maybe other people have stronger opinions (@AlexanderLyNL ?)

Reply:
Indeed, the problem is partial matching and can result in some obscure errors. Partial matching makes R more flexible, but sometimes too flexible, which is why I discourage it in packages.

Reply:
Ah, ok, that makes sense. Thanks for the explanation!